src/System.IO.Compression.Brotli/System/IO/Compression/Brotli.cs

```diff
@@ -167,7 +181,14 @@ public static TransformationStatus FlushEncoder(Span<byte> source, Span<byte> de
 
         public static TransformationStatus Compress(ReadOnlySpan<byte> source, Span<byte> destination, out int bytesConsumed, out int bytesWritten, ref State state)
         {
-            EnsureInitialized(ref state, true);
+            try
```

Comment:
This try catch and throw is not necessary if all you are doing it re-throwing the same exception. Just let it bubble up to the caller.